app/assets/javascripts/discourse-common/addon/lib/raw-templates.js

```diff
@@ -31,17 +31,18 @@ export function findRawTemplate(name) {
   );
 }
 
-export function buildRawConnectorCache(findOutlets) {
+export function buildRawConnectorCache() {
   let result = {};
-  findOutlets(
-    Object.keys(__DISCOURSE_RAW_TEMPLATES),
-    (outletName, resource) => {
+  Object.keys(__DISCOURSE_RAW_TEMPLATES).forEach((resource) => {
+    const segments = resource.split("/");
+    if (segments.includes("connectors")) {
+      const outletName = segments[segments.length - 2];
```

Comment:
Bear with me as I might mess part of the problem here. But what about:

```
const segments = resource.split("/");
const connectorIndex = segments.indexOf("connectors")

if (connectorIndex) {
  const outletName = segments[connectorIndex + 1];
  // ...
}
```

There's the risk of out of range in unexpected input, probably possible on the previous code too, but it seems more resilient to me maybe?

Reply:
I saw the regex after, it's probably not possible to end up in a situation where being resilient is necessary :+1:

Reply:
Looks good - intention is much clearer, thanks!